### PR TITLE
[Tool] Update tool chain and dev env dockerfiles

### PR DESCRIPTION
**Pull request by fieldsfarmer**
## Why I'm doing:
We need to download some artifacts from github private repositories. Meanwhile we wish to align our code with StarRocks (instead of using our own docker file). This change could also help other teams as they are quite general.

## What I'm doing:
- Install the github command line `gh` in the tool chain image docker file to help us make github related operations more easily.
- Add an argument `GITHUB_TOKEN` in the dev env image docker file to help us download artifacts in github private repositories.

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: docker/dockerfiles/dev-env/dev-env.Dockerfile [hunk 1]
```diff
@@ -25,6 +25,8 @@ ARG commit_id
 ARG starlet_tag=v3.2-rc8
 # build for which linux distro: centos7|ubuntu
 ARG distro=ubuntu
+# Token to access artifacts in private github repositories.
+ARG GITHUB_TOKEN
```
Comment (chaoyli) [inline docker/dockerfiles/dev-env/dev-env.Dockerfile:29]: How to use the parameter?

Comment (fieldsfarmer) [reply docker/dockerfiles/dev-env/dev-env.Dockerfile:29]: DONE

File: docker/dockerfiles/toolchains/toolchains-ubuntu.Dockerfile [hunk 2]
```diff
@@ -7,7 +7,7 @@ FROM ubuntu:22.04
 RUN apt-get update -y && \
     apt-get install --no-install-recommends -y \
     automake binutils-dev bison byacc ccache flex libiberty-dev libtool maven zip python3 python-is-python3 make cmake gcc g++ default-jdk git patch lld bzip2 \
-    wget unzip curl vim tree net-tools openssh-client xz-utils && \
+    wget unzip curl vim tree net-tools openssh-client xz-utils gh && \
```
Comment (kevincai) [inline docker/dockerfiles/toolchains/toolchains-ubuntu.Dockerfile:10]: this makes the gh only works for ubuntu platform, not working for centos7. possibly worth comment that it won't work for centos7 at all.

Comment (fieldsfarmer) [reply docker/dockerfiles/toolchains/toolchains-ubuntu.Dockerfile:10]: Added in centos7 docker file to install gh. Tested locally.

**Review by kevincai — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by chaoyli** [comment]
@mergify backport 3.1 3.2

**Comment by chaoyli** [comment]
@mergify backport branch-3.1 branch-3.2

**Pull request merged**